### PR TITLE
Use our Vagrantfile during tests

### LINKED ISSUE #202: Use our Vagrantfile with test kitchen
We currently use kitchen-ansible to run the playbook, but this leads to a difference in how we provision and configure the environment, and it requires that we first run the test kitchen suites, and then run our Vagrantfile.

We can customize the Vagrantfile template that kitchen-vagrant uses to run our Vagrantfile instead.

### PR DESCRIPTION
**Pull request by ferrarimarco**
This PR does the following:

1. Implements a testing step during the CI build that uses our `Vagrantfile` to build the base box.
1. Runs vagrant validate when building the base box.
1. Removes the vagrant validation step in the linting stage.
1. Removes Test-Kitchen, kitchen-vagrant, kitchen-ansible, and kitchen-inspec from the toolchain. We still use InSpec as verifier.
1. Caches both gems installed by bundler and by Vagrant, shortening the build by two minutes.
1. Adds more debugging output during the CI build.
1. Fixes an initialization warning of constants in Vagrantfile.
1. Implements a function to allow suppressing output in Vagrantfile.
1. Adds missing metadata to InSpec profiles and controls.

Close #202

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ dist: bionic
 env:
   global:
     - VAGRANT_VERSION="2.2.7"
+    - BASE_BOX_VAGRANT_VM_ID="base-box-builder.k8s-play.local"
```
Comment (StefanoSalsano) [inline .travis.yml:6]: testbed name is configurable in defaults.yaml
`playground_name: k8s-play`

File: Vagrantfile [hunk 2]
```diff
@@ -138,16 +144,16 @@ if File.exist?(env_specific_config_path)
 end
 
 # Display the main current configuration parameters
-@ui.info "Welcome to Kubernetes playground!"
-@ui.info "Vagrant provider: " + settings["conf"]["vagrant_provider"]
+output_info_if_enabled "Welcome to Kubernetes playground!"
```
Comment (StefanoSalsano) [inline Vagrantfile:147]: not a great concern, but a better and shorter name could be
`ui_info_if_enabled`
or simply
`ui_info`

File: .travis.yml (hunk 1 above)
Comment (ferrarimarco) [reply .travis.yml:6]: fixed. Set a value in the env.yaml that we generate for travis

File: Vagrantfile (hunk 2 above)
Comment (ferrarimarco) [reply Vagrantfile:147]: Ok for ui_info_if_enabled

**Pull request merged**